Support ingesting data for environments with Identity

Create lean environment with Identity Enabled. 
Add following paramters: 
INGEST_ACCESS_CONTROL
INGEST_CUSTOM_SERVICE_AGREEMENTS
INGEST_TRANSACTIONS
INGEST_APPROVALS_FOR_PAYMENTS
INGEST_PAYMENTS
IDENTITY_FEATURE_TOGGLE : Enable

and build.

The build always fails with following error :

`Request method:	POST
Request URI:	http://edge.9455.dbs.backbase.test/api/accessgroup-presentation-service/client-api/v2/accessgroups/usercontext?_csrf
Proxy:			<none>
Request params:	<none>
Query params:	_csrf
Form params:	<none>
Path params:	<none>
Headers:		Accept=*/*
				Content-Type=application/json; charset=UTF-8
Cookies:		Authorization=eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJoWHhHQm9SNUtSdW0ybFhNcnRaUGFxNF9GRWNISnBpM0dFS0pnd21pbWRBIn0.eyJleHAiOjE1OTMxNjg4MzUsImlhdCI6MTU5MzE2ODUzNSwianRpIjoiNmRhYmMyZTEtYTNkYS00OWY2LTljODgtZTJjNTc5OTIxMmQ4IiwiaXNzIjoiaHR0cDovL2lkZW50aXR5Ljk0NTUuZGJzLmJhY2tiYXNlLnRlc3QvYXV0aC9yZWFsbXMvYmFja2Jhc2UiLCJhdWQiOiJhY2NvdW50Iiwic3ViIjoiNjYyOWRhNmQtMzY5Zi00OWEzLTgzZDAtMmFlOGJkNzZkZTRiIiwidHlwIjoiQmVhcmVyIiwiYXpwIjoiYmItdG9vbGluZy1jbGllbnQiLCJzZXNzaW9uX3N0YXRlIjoiYTU4ZWZlZmEtNTMwOS00ZDhkLTliODctNjJjM2VjMzJiNDFmIiwiYWNyIjoiMSIsImFsbG93ZWQtb3JpZ2lucyI6WyIqIl0sInJlYWxtX2FjY2VzcyI6eyJyb2xlcyI6WyJST0xFX0FDVFVBVE9SIiwiUk9MRV9BRE1JTiIsIlJPTEVfZ3JvdXBfYWRtaW4oQURNSU4pIl19LCJyZXNvdXJjZV9hY2Nlc3MiOnsiYWNjb3VudCI6eyJyb2xlcyI6WyJtYW5hZ2UtYWNjb3VudCIsInZpZXctcHJvZmlsZSJdfX0sInNjb3BlIjoiZW1haWwgcHJvZmlsZSIsImVtYWlsX3ZlcmlmaWVkIjpmYWxzZSwidXNlcl9uYW1lIjoiYWRtaW4iLCJuYW1lIjoiQWRtaW4gdXNlciBhZG1pbmlzdHJhdG9yIiwicHJlZmVycmVkX3VzZXJuYW1lIjoiYWRtaW4iLCJnaXZlbl9uYW1lIjoiQWRtaW4gdXNlciIsImZhbWlseV9uYW1lIjoiYWRtaW5pc3RyYXRvciIsImF1dGhvcml0aWVzIjpbIlJPTEVfQUNUVUFUT1IiLCJST0xFX0FETUlOIiwiUk9MRV9ncm91cF9hZG1pbihBRE1JTikiXX0.aRh_SpPDKndkUvRHcjHulUT-C9IO2gPzPha1oU2OtWoCi5xegYNvcde9QwATZYeQtAa9_dfHUP6qV6u7Y2K3_8IdCvI2EHCGzqV-enn-kf86S8I6HDte198Ln09CzL0GK2AuxsH-RhYPaANgrk9gX-cBh7BTMkEmNWIUfVe718O2Rk94GZiOWdYpUHUqmRlN-jcwcd2gOJVv9sXmoV6WphLVS8WQZPojsElMzUrDefmQ1hl3eq2CKblV4ePL3DaeUF9G7EzKi7mTU6zUyS-KvZUKYQKcP8aHAPdkk0y_B_clYltpfx_TLRGbDVVn1XdVfcXQmKK_OlQT1xcgU055SA
				KEYCLOAK_LOCALE=
				KC_RESTART=
Multiparts:		<none>
Body:
{
    "serviceAgreementId": "8a49b7eb72f0329e0172f03e337f0001"
}
HTTP/1.1 403 Forbidden
Server: nginx/1.17.8
Date: Fri, 26 Jun 2020 10:48:57 GMT
Content-Type: application/json
Transfer-Encoding: chunked
Connection: keep-alive
Vary: Accept-Encoding
X-Xss-Protection: 1 ; mode=block
Strict-Transport-Security: max-age=31536000; includeSubDomains
X-Frame-Options: SAMEORIGIN
X-Content-Type-Options: nosniff
Referrer-Policy: no-referrer
X-Download-Options: noopen
X-Permitted-Cross-Domain-Policies: none
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: 0
X-Frame-Options: DENY
Set-Cookie: XSRF-TOKEN=f6d5d91f-3596-46ab-b4b1-6d4fec006f62; Path=/
Content-Encoding: gzip
{
    "timestamp": "2020-06-26T10:48:57.869+0000",
    "status": 403,
    "error": "Forbidden",
    "message": "Forbidden",
    "path": "/client-api/v2/accessgroups/usercontext"
}
2020-06-26 12:48:57.948 ERROR [bb-fuel,,,] 24801 --- [           main] o.s.boot.SpringApplication               : Application run failed
java.lang.AssertionError: 1 expectation failed.
Expected status code <204> but was <403>.`

--- a/src/main/java/com/backbase/ct/bbfuel/client/common/LoginRestClient.java
+++ b/src/main/java/com/backbase/ct/bbfuel/client/common/LoginRestClient.java
@@ -1,11 +1,6 @@
 package com.backbase.ct.bbfuel.client.common;
 
-import static com.backbase.ct.bbfuel.data.CommonConstants.ACCESS_TOKEN;
-import static com.backbase.ct.bbfuel.data.CommonConstants.IDENTITY_AUTH;
-import static com.backbase.ct.bbfuel.data.CommonConstants.IDENTITY_TOKEN_PATH;
-import static com.backbase.ct.bbfuel.data.CommonConstants.PROPERTY_IDENTITY_CLIENT;
-import static com.backbase.ct.bbfuel.data.CommonConstants.PROPERTY_IDENTITY_FEATURE_TOGGLE;
-import static com.backbase.ct.bbfuel.data.CommonConstants.PROPERTY_IDENTITY_REALM;
+import static com.backbase.ct.bbfuel.data.CommonConstants.*;
 import static org.apache.http.HttpStatus.SC_OK;
 
 import com.backbase.ct.bbfuel.config.BbFuelConfiguration;
@@ -55,9 +50,13 @@ public class LoginRestClient extends RestClient {
                 .statusCode(SC_OK);
 
             String token = response.extract().jsonPath().get(ACCESS_TOKEN);
+            String refreshToken = response.extract().jsonPath().get(REFRESH_TOKEN);
+            String csrfToken = response.extract().jsonPath().get(SESSION_TOKEN);
 
             cookies = new HashMap<>(response.extract().cookies());
             cookies.put("Authorization", token);
+            cookies.put("refresh_token", refreshToken);
+            cookies.put("XSRF-TOKEN", csrfToken);
 
         } else {
             response = requestSpec().param("username", username)

--- a/src/main/java/com/backbase/ct/bbfuel/client/common/LoginRestClient.java
+++ b/src/main/java/com/backbase/ct/bbfuel/client/common/LoginRestClient.java
@@ -1,6 +1,13 @@
 package com.backbase.ct.bbfuel.client.common;
 
-import static com.backbase.ct.bbfuel.data.CommonConstants.*;
+import static com.backbase.ct.bbfuel.data.CommonConstants.ACCESS_TOKEN;
+import static com.backbase.ct.bbfuel.data.CommonConstants.IDENTITY_AUTH;
+import static com.backbase.ct.bbfuel.data.CommonConstants.IDENTITY_TOKEN_PATH;
+import static com.backbase.ct.bbfuel.data.CommonConstants.PROPERTY_IDENTITY_CLIENT;
+import static com.backbase.ct.bbfuel.data.CommonConstants.PROPERTY_IDENTITY_FEATURE_TOGGLE;
+import static com.backbase.ct.bbfuel.data.CommonConstants.PROPERTY_IDENTITY_REALM;
+import static com.backbase.ct.bbfuel.data.CommonConstants.REFRESH_TOKEN;
+import static com.backbase.ct.bbfuel.data.CommonConstants.SESSION_TOKEN;
 import static org.apache.http.HttpStatus.SC_OK;
 
 import com.backbase.ct.bbfuel.config.BbFuelConfiguration;

--- a/src/main/java/com/backbase/ct/bbfuel/data/CommonConstants.java
+++ b/src/main/java/com/backbase/ct/bbfuel/data/CommonConstants.java
@@ -100,4 +100,6 @@ public final class CommonConstants {
     public static final String IDENTITY_AUTH = "/auth/realms";
     public static final String IDENTITY_TOKEN_PATH = "/protocol/openid-connect/token";
     public static final String ACCESS_TOKEN = "access_token";
+    public static final String REFRESH_TOKEN = "refresh_token";
+    public static final String SESSION_TOKEN = "session_state";
 }

